contracts/SafeMath.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.7;
+
+/**
+ * @dev Wrappers over Solidity's arithmetic operations with added overflow
+ * checks.
+ *
+ * Arithmetic operations in Solidity wrap on overflow. This can easily result
+ * in bugs, because programmers usually assume that an overflow raises an
+ * error, which is the standard behavior in high level programming languages.
+ * `SafeMath` restores this intuition by reverting the transaction when an
+ * operation overflows.
+ *
+ * Using this library instead of the unchecked operations eliminates an entire
+ * class of bugs, so it's recommended to use it always.
+ */
+library SafeMath {
```

Comment:
Seems like there's a lot of places we could still use this but don't. Any particular reason why we do/don't use it in different places?

Reply:
Good catch; we should remove this library. We're using solc >0.8 now which by default reverts on arithmetic errors. So `SafeMath` has been removed as recommended [here](https://docs.soliditylang.org/en/v0.8.7/080-breaking-changes.html).